pkg/cmd/listen.go

```diff
@@ -114,7 +117,7 @@ func (lc *listenCmd) runListenCmd(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	ctx := withSIGTERMCancel(cmd.Context(), func() {
+	ctx := withSIGTERMCancel(lc.parentContext, func() {
```

Comment:
note: if we simply replace `cmd.Context()` with `context.Background()` then there is no crash in playback as well. However, #747 makes it seems that if we can pass context from a parent, we should